docs/advanced/scopes-loadcontexts.rst

```diff
@@ -0,0 +1,75 @@
+AssemblyLoadContext and Lifetime Scopes
+=======================================
+
+In .NET Core and .NET 5+, the `the AssemblyLoadContext was introduced <https://learn.microsoft.com/en-us/dotnet/core/dependency-loading/understanding-assemblyloadcontext>`_.
+
+This allowed developers to dynamically load and unload assemblies from their application, and is very useful for developers writing applications with plugin-based architectures.
+
+In order to unload the assemblies in an ``AssemblyLoadContext`` when a plugin is no longer needed, and reclaim the memory required by that plugin, no references to types in that assembly can be held by anything outside the loaded context. This includes Autofac, which by default holds a variety of internal references and caches for types that have been registered.
+
+As of Autofac 7.0, we've added support for indicating to Autofac that a given lifetime scope represents types loaded for a given ``AssemblyLoadContext``; when a lifetime scope created for a specific ``AssemblyLoadContext`` is unloaded, Autofac will make a *best-effort* attempt to remove all references we hold for types from the loaded context, to allow the ``AssemblyLoadContext`` to be unloaded.
+
+You indicate that a lifetime scope is for an ``AssemblyLoadContext`` with the new ``BeginLoadContextLifetimeScope`` method. Here's a full example:
+
+.. sourcecode:: csharp
+
+    //
+    // In PluginDefinition project.
+    //
+
+    public interface IPlugin
+    {
+      void DoSomething();
+    }
+
+    //
+    // In MyPlugin project.
+    //
+
+    public class MyPlugin : IPlugin
+    {
+        public void DoSomething()
+        {
+          Console.WriteLine("Hello World");
+        }
+    }
+
+    //
+    // In the application.
+    //
+
+    var builder = new ContainerBuilder();
+
+    // Components defined in the "Default" load context will be available in load context lifetime scopes.
+    builder.RegisterType<DefaultComponent>();
+
+    var container = builder.Build();
+
+    var loadContext = new AssemblyLoadContext("PluginContext", isCollectible: true);
+
+    using (var scope = container.BeginLoadContextLifetimeScope(loadContext, builder =>
+    {
+      var pluginAssembly = loadContext.LoadFromAssemblyPath("plugins/MyPlugin.dll");
+
+      builder.RegisterAssemblyTypes(pluginAssembly);
+    }))
+    {
+      // Because of default load context behaviour, because we have already loaded
+      // PluginDefinition into the *application* so we can reference IPlugin, the loaded MyPlugin assembly
+      // will share the same type.
```

Comment:
```suggestion
      // The application should reference the PluginDefinition project, which means the
      // default load context will have loaded the IPlugin interface already. When the
      // MyPlugin assembly gets loaded it should share the same type and allow resolution
      // with the common interface.
```

I think this is what you were going for but some of the "because..., because..." got a bit confusing for me.